src/build/config/config.yaml

```diff
@@ -2423,6 +2431,19 @@ option:
     command: repo-type
     depend: repo-sftp-host
 
+  repo-sftp-strict-host-key-checking:
+    section: global
```

Comment:
@sfrost I would prefer this to be `repo-sftp-strict-host-key-check` to be more consistent with our option naming, but I realize this name was selected to match up with the OpenSSH setting. What do you think?

Reply:
I'm fine with either naming.  More concerned about people being able to understand quickly how it works than about the exact name, which is never going to match what the OpenSSH config option is named anyway.

Reply:
I have no objection to changing it.

Reply:
OK, renamed, but still not too satisfied. The option is one we'd usually end with "-type", which I can live with, but it also appears that repo-sftp-strict-host-key-check depends on whether repo-sftp-host-fingerprint is specified, which is exactly backward from the way we usually do it.

I'm guessing this was done to (try to) avoid breakage, but it's not ideal. Usually the way we would handle this is to have a type (which is default) that enables repo-sftp-host-fingerprint and other types that enable repo-sftp-known-host. That preserves the current behavior, but my guess is you want to auto change the current behavior, which may cause breakage anyway if they do not have repo-sftp-host-fingerprint set.

Thoughts? Am I missing something?

Reply:
I'd much prefer to have the known-host checking stuff get automatically picked up, if no fingerprint has been set (if a fingerprint has been set ... then we haven't really got much business groveling around through known hosts files, imv).

By default, ssh has 'ask' for StrictHostKeyChecking which will ask about not-yet-known hosts, but for hosts who have a known fingerprint will refuse to connect if the fingerprint from the server doesn't match what's in the known hosts.  This is basically what we want to get to and yes, that does mean we might break things for people who didn't specify a fingerprint and who don't have the host known yet in known-hosts.

Having an option for repo-sftp-strict-host-key-checking which is "use the provided fingerprint" could be alright but seems a bit silly- if it's provided, we should use it, not ignore it, regardless of what this other option is set to.  Still, I won't push back too hard if that's what you want to do.  What I would want, however, is a way to set the *default* for repo-sftp-strict-host-key-checking to be "use the provided fingerprint, but if there isn't one, then use the known hosts file".  This all seems like it's getting pretty far from how that option works in ssh_config though and I'm not super thrilled with that.

Hopefully this at least helps clear up some of the intention here.  I'm open to different ways of getting there.

Reply:
OK, let me think on it. I'm not very content with the dependency checking being backward -- and with no dependencies in the config. We don't have any way for a type option to depend on a string because we don't do it that way in general...

Reply:
I believe I have resolved this in 672a3130c5. Basically, update the dependencies to be the (mostly) correct but add some rules in `cfgLoadUpdateOption()` to error on new cases where `repo-sftp-strict-host-key-check` is explicitly set incorrectly, but allow the current config of fingerprint to pass with a warning to update the config.

Reply:
@sfrost I believe we have an unresolved issue from our discussion about whether to rename `repo-sftp-strict-host-key-check` to something like `repo-sftp-host-key-check-type`? I guess the values would be `strict`, `fingerprint`, accept-new` and `none`?

Reply:
I'm fine with the rename.